src/misc_early.rs

```diff
@@ -113,4 +125,15 @@ impl EarlyLintPass for MiscEarly {
             }
         }
     }
+
+    fn check_expr(&mut self, cx: &EarlyContext, expr: &Expr) {
+        if let ExprKind::Call(ref paren, _) = expr.node {
+            if let ExprKind::Paren(ref closure) = paren.node {
+                if let ExprKind::Closure(_, _, _) = closure.node {
+                    cx.span_lint(CLOSURE_CALLED_EARLY, expr.span, "Try not to call \
+                        a closure in the expression where it is declared.")
```

Comment:
I'd like to see a [span_suggestion](http://manishearth.github.io/rust-internals-docs/syntax/errors/struct.DiagnosticBuilder.html#method.span_suggestion) here, which suggests `(<closure contents>)` if the closure has no arguments.

Provide the span of the call, and use `snippet()` on the block inside the closure to obtain the suggestion (see other instances of span_suggestion in clippy for examples)